File: snippets/rs/array/cut/cut.rs
fn main()
{
    let mut array = vec![5, 4, 3, 2, 1];
    let start = 1;  // 0: first item, 1: second item...
    let count = 3;
    println!("{array:?}");      // [5, 4, 3, 2, 1]
    let array_cut: Vec<_> = array.drain(start..(start + count)).collect();  // ≡
    println!("{array:?}");      // [5, 1]
    println!("{array_cut:?}");  // [4, 3, 2]
}

File: snippets/rb/array/cut/cut.rb
array = [5, 4, 3, 2, 1]
start = 1  # 0: first item, 1: second item...
count = 3
print(array)     # [ 5, 4, 3, 2, 1 ]
print("\n")
arrayCut = array[start,(start + count - 1)] 
array.slice!(start,(start + count - 1))
print(array)     # [ 5, 1 ]
print("\n")
print(arrayCut)  # [ 4, 3, 2 ]